pom.xml

```diff
@@ -558,6 +558,15 @@
             <groupId>org.apache.maven.plugins</groupId>
             <artifactId>maven-gpg-plugin</artifactId>
             <version>1.6</version>
+            <configuration>
+              <useAgent>true</useAgent>
+              <passphrase>${env.GPG_PASSPHRASE}</passphrase>
+              <gpgArguments>
+                <arg>--batch</arg>
```

Comment:
With maven-gpg-plugin 1.6, batch is added to the gpg command generated. So you can then remove it - see code: https://github.com/apache/maven-gpg-plugin/blob/maven-gpg-plugin-1.6/src/main/java/org/apache/maven/plugin/gpg/GpgSigner.java#L93-L100

you can test it locally
```
echo ${GPG_PASSPHRASE} | gpg --batch --passphrase-fd 0 --pinentry-mode loopback --local-user ${KEYNAME} --armor --detach-sign --no-default-keyring --output target/pom.xml.asc pom.xml
```

Reply:
Done.